Correct `$(SharedFxVersion)` and `$(TargetingPackVersion)` values

dougbu opened the pull request:
- Ensure `$(SharedFxVersion)` doesn't change in `$(NoSemVer20)` projects
- Ignore current project's `$(VersionSuffix)` in `$(TargetingPackVersion)`
    - Never assume `$(AspNetCoreBaselineVersion)` matches released targeting pack
- Stabilize both versions correctly
- Use these properties more widely
    - Remove other mechanisms to get the same values
    - Reduce use of the `_GetPackageVersionInfo` target
    - Reduce use of `$(SharedFxVersion)` for the targeting pack

nits:
- Correct comments about old RTMVersions.csproj project
- Fix or remove a few other comments

dougbu committed:
Correct `$(SharedFxVersion)` and `$(TargetingPackVersion)` values
- Ensure `$(SharedFxVersion)` doesn't change in `$(NoSemVer20)` projects
- Ignore current project's `$(VersionSuffix)` in `$(TargetingPackVersion)`
    - Never assume `$(AspNetCoreBaselineVersion)` matches released targeting pack
- Stabilize both versions correctly
- Use these properties more widely
    - Remove other mechanisms to get the same values
    - Reduce use of the `_GetPackageVersionInfo` target
    - Reduce use of `$(SharedFxVersion)` for the targeting pack

nits:
- Correct comments about old RTMVersions.csproj project
- Fix or remove a few other comments

dougbu commented:
This is more work coming out of our servicing readiness exercise. My previous changes to use `$(SharedFxVersion)` more in our Helix setup were useful but didn't go quite far enough. This got those builds much further❕ (Then I needed to back-port too many build reliability issues ☹️)

dougbu commented:
@Pilchie any objection to tell-mode for RC2❔

HaoK commented on eng/targets/Helix.targets:
```diff
@@ -106,14 +106,8 @@ Usage: dotnet msbuild /t:Helix src/MyTestProject.csproj
-    -->
     <ItemGroup Condition=" '$(TestDependsOnAspNetRef)' == 'true' AND '$(IsTargetingPackBuilding)' == 'true' ">
-      <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\Microsoft.AspNetCore.App.Ref.$(SharedFxVersion).nupkg" />
+      <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\Microsoft.AspNetCore.App.Ref.$(TargetingPackVersion).nupkg" />
```
Nit: can we put this in a variable to use in both here and the nupkg file we pass below into the commands?   
`Microsoft.AspNetCore.App.Ref.$(TargetingPackVersion).nupkg`

dougbu replied:
Looks like I need to rebase anyhow. So, sure, I'll make another update here.

But, I'll go w/ your earlier suggestion of removing the `*.nupkg` names from the command line and instead use the passed `$(SharedFxVersion)` to craft the names on the Helix agent.

HaoK replied:
Nice, a few less arguments is much better since its already getting out of control with how many we pass!

Pilchie commented:
Tell mode is fine with me.

JamesNK commented:
Hmmm, this might be related to some failures in the gRPC interop tests here - https://github.com/dotnet/aspnetcore/pull/25587

Offline discussion with @JunTaoLuo:

> I'm trying to update the gRPC template with a new package version for RC2. It uses new fields added to HeaderNames and the interop tests are failing because they not in the aspnetcore runtime that it is using. This is blocking me from merging for RC2.
> 
> The test writes assembly versions in the logs and this is what the server reports:
> 
>   NetCoreAppVersion: 5.0.0-rc.1.20371.13+d8cf13e0ba9b369a15a83472b6b97463c6d07fe2
>   AspNetCoreAppVersion: 5.0.0-rc.1.20379.2
> 
> Oddly, the client writes a different version:
> 
>   NetCoreAppVersion: 5.0.0-rc.2.20458.14+482494f9ecc9a630599278b80893841462c90a5b
> 
> It looks like the server is using a wrong version. It reports RC1 and this PR is targeting the RC2 branch.

Possible fix John added to that PR - [`29a835f` (#25587)](https://github.com/dotnet/aspnetcore/pull/25587/commits/29a835f7a41ddccae69ad3e26f2b0f95c4053a8c)

Could this issue be the cause instead?

dougbu committed:
Do not pass package filenames on Helix command Lines
- remove parsing of these command-line arguments from `RuntestOptions`
  - instead craft the names using passed `$(SharedFxVersion)`
- restore `$(DotNetRuntimeSourceFeedKey)` on Helix command line
  - lost somewhere along the line
- correct argument count in runtests.sh
  - treated 11th argument as both Helix timeout and feed credential
  - count was messed up somewhere alone the line

nits:
- update C# syntax in `RuntestOptions` e.g. remove unused `public` setters
- sort and group properties and their assignments

dougbu commented:
> Could this issue be the cause instead?

It's possible but somewhat unlikely. The two property values I'm setting here were wrong for a small subset of the projects and mainly when we're using stable versions. We're not using stable versions (outside a servicing readiness exercise) for 5.0 yet. So, this PR should have minimal impact -- for now.

The main exception is if we needed the shared Fx or targeting pack version in a project using SemVer 1.0. So far, we're :ok: there.

So, @JunTaoLuo's fix in your PR is on the right path. Any place we made assumptions that dotnet/runtime and dotnet/aspnetcore sharedFx or TFM and dotnet/aspnetcore sharedFx versions align is usually wrong.

dougbu commented:
@HaoK @wtgodbe please have another look. The second commit changes things from what you saw when commenting.

@Pilchie any objection if I get this in tonight or over the weekend❔

Pilchie commented:
No objection!
